crates/router/src/connector/shift4/transformers.rs

```diff
@@ -157,66 +169,61 @@ fn get_card_payment_request<T>(
     item: &types::RouterData<T, types::PaymentsAuthorizeData, types::PaymentsResponseData>,
     card: &api_models::payments::Card,
 ) -> Result<Shift4PaymentsRequest, Error> {
-    let submit_for_settlement = item.request.is_auto_capture()?;
+    let captured = item.request.is_auto_capture()?;
     let card = Card {
         number: card.card_number.clone(),
         exp_month: card.card_exp_month.clone(),
         exp_year: card.card_exp_year.clone(),
         cardholder_name: card.card_holder_name.clone(),
     };
-    if item.is_three_ds() {
-        Ok(Shift4PaymentsRequest::ThreeDSRequest(Box::new(
-            Shift43DSRequest {
-                amount: item.request.amount.to_string(),
-                currency: item.request.currency.to_string(),
-                card_number: card.number,
-                card_exp_month: card.exp_month,
-                card_exp_year: card.exp_year,
-                return_url: item
-                    .request
-                    .complete_authorize_url
-                    .clone()
-                    .ok_or_else(|| errors::ConnectorError::RequestEncodingFailed)?,
-            },
-        )))
+    let amount = item.request.amount.to_string();
+    let currency = item.request.currency;
+    let payment_method = if item.is_three_ds() {
+        Shift4PaymentMethod::Cards3DSRequest(Box::new(Cards3DSRequest {
+            card_number: card.number,
+            card_exp_month: card.exp_month,
+            card_exp_year: card.exp_year,
+            return_url: item
+                .request
+                .complete_authorize_url
+                .clone()
+                .ok_or_else(|| errors::ConnectorError::RequestEncodingFailed)?,
+        }))
     } else {
-        Ok(Shift4PaymentsRequest::Non3DSRequest(Box::new(
-            Shift4Non3DSRequest {
-                amount: item.request.amount.to_string(),
-                card: Some(CardPayment::RawCard(Box::new(card))),
-                currency: item.request.currency.to_string(),
-                description: item.description.clone(),
-                captured: submit_for_settlement,
-                payment_method: None,
-                flow: None,
-            },
-        )))
-    }
+        Shift4PaymentMethod::CardsNon3DSRequest(Box::new(CardsNon3DSRequest {
+            card: CardPayment::RawCard(Box::new(card)),
+            description: item.description.clone(),
+        }))
+    };
+    Ok(Shift4PaymentsRequest {
```

Comment:
Since we moved the payment method data into enum's, there is no need to return entire Shift4PaymentsRequest in `get_card_payment_request` & `get_bank_redirect_request` request. We should return only Shift4PaymentMethod from these method. 
In fact we should move from `get_card_payment_request` & `get_bank_redirect_request` methods to try_from implementation, since these are type conversion.